LFS171x/docs/introduction-to-hyperledger-composer.md

```diff
@@ -0,0 +1,976 @@
+
+Cloud providers often provide free credits or have plans enabling free usage of small VMs
+
+### Text editor
+On your local machine, we recommend you have an editor featuring a plugin for Hyperledger Composer.
+
+Two such editors exist today:
+
+* [Atom](https://atom.io/)
+* [VS Code](https://code.visualstudio.com)
+
+These will enable you to more easily work using the Hyperledger Composer Extension.
+
+
+### Installing prerequisites
+
+Connect to the command line of your virtual machine (e.g. by using SSH).
+
+Download and install the prerequisites:
+
+```bash
+curl -O https://hyperledger.github.io/composer/latest/prereqs-ubuntu.sh
+chmod u+x prereqs-ubuntu.sh
+./prereqs-ubuntu.sh
+```
+
+## Technical Prerequisites - Mac OS
+
+To install Hyperledger Composer in your Mac you can either follow this video or the guided the procedure in the official [Hyperledger Composer Documentation](https://hyperledger.github.io/composer/latest/installing/installing-prereqs.html#macos).
+
+> **VIDEO [EdX_1_prerequisites.mp4]**
+> Transcript:
+
+> In this video we will install the pre-requisites on a Mac.
+>
+> Let us open up a Terminal window. We first check if XCode is installed, which in our case it is. If it is not installed on your computer, you will be prompted to install it in the next step.
+>
+> Now we install the Node Version Manager or nvm. You may need to reload your bash shell configuration file. Now we can install the LTS version of node.js and activate it.
+>
+> We now download and install the Docker Community Edition. You may need to make an account. We drag Docker into our Applications folder. Now open it and login with your account. Check that it works by running the hello-world image.
+>
+> We now download the VSCode editor. We drag it to our Applications Folder and open it. We go to the Extension section, and search for Hyperledger. We then install the Hyperledger Composer extension.
+>
+> Let us switch back to the terminal. In our bash configuration file we tell NVM to use the LTS version of Node. And we can now use the Node Package Manager to install the Hyperledger Composer tools.
+>
+> After this we download the Hyperledger Fabric development server scripts. We extract the files and we run downloadFabric.sh to fetch the Docker images.
+>
+> We can then start the Hyperledger Fabric development server and create a peerAdmin identity card.
+>
+> Finally, we start the Hyperledger Composer playground, which we will use in a subsequent video.
+
+### Text editor
+On your local machine, we recommend you have an editor featuring a plugin for Hyperledger Composer.
+
+Two such editors exist today:
+
+* [Atom](https://atom.io/)
+* [VS Code](https://code.visualstudio.com)
+
+These will enable you to more easily work using the Hyperledger Composer Extension.
+
+## Installing Hyperledger Composer Components
+
+You will now need to install the Hyperledger Composer components:
+
+This installs the Composer Command Line Interface (CLI)
+
+```
```

Comment:
When running this after installing the pre-requisites on my Linux virtualbox VM, I get:
```
The program 'npm' is currently not installed. You can install it by typing:
sudo apt install npm
```

I had to do the following before it would work:
```
$ export NVM_DIR="${HOME}/.nvm"
$ [ -s "${NVM_DIR}/nvm.sh" ] && . "${NVM_DIR}/nvm.sh"
$ [ -s "${NVM_DIR}/bash_completion" ] && . "${NVM_DIR}/bash_completion"
$ nvm use --lts
Now using node v8.11.4 (npm v6.4.0)
$ npm install -g composer-cli
```

Reply:
Looks like this was also caused by not re-logging in. Probably worth a bold note in the Linux section on re-logging in.

Reply:
Added a notice in the pre-requisites part mentioning that one may need to log out and back into the VM.